### PR TITLE
OSD-6497 dont issue certs during relocation

### PR DESCRIPTION
**Pull request by yithian**

### CODE REVIEW

**Comment by yithian** [comment]
this is based on #176 and will block on it

File: pkg/controller/certificaterequest/certificaterequest_controller.go [hunk 1]
```diff
@@ -136,6 +138,20 @@ func (r *ReconcileCertificateRequest) Reconcile(request reconcile.Request) (reco
...
+	// fetch the clusterdeployment and bail out if there's an outgoing migration annotation
+	cd := &hivev1.ClusterDeployment{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: request.Namespace, Name: strings.Split(request.Name, fmt.Sprintf("-%s", strings.Split(cr.Spec.CertificateSecret.Name, "-secret")[0]))[0]}, cd)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// bail out of the loop if there's an outgoing relocation annotation
+	for a, v := range cd.Annotations {
+		if a == "hive.openshift.io/relocate" && strings.Split(v, "/")[1] == "outgoing" {
```
Comment (dustman9000) [inline pkg/controller/certificaterequest/certificaterequest_controller.go:150]: can we turn these strings into constants?

File: pkg/controller/certificaterequest/certificaterequest_controller.go [hunk 2]
```diff
@@ -136,6 +138,20 @@ func (r *ReconcileCertificateRequest) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
+	// fetch the clusterdeployment and bail out if there's an outgoing migration annotation
```
Comment (dustman9000) [inline pkg/controller/certificaterequest/certificaterequest_controller.go:141]: can we separate out the formatting of the cd name to make this a little more readable?

**Pull request merged**